### DISCUSSION TITLE
fallo en el modulo 1 primeros pasos en "calculadora"

### DISCUSSION THREAD

**Discussion by larap03** [LaunchX-InnovaccionVirtual/CursoIntroPython#143] [Q&A]
esta bien hasta que cambie de laptop
![image](https://user-images.githubusercontent.com/99061466/153653049-613c2335-ddb3-4c1a-9fb2-b59686a5d788.png)

**Comment by enriqueorea** [comment]
Esta declarando mal tus variables, recuerda que la función input muestra un mensaje en terminal para ingresar un valor
la forma correcta "primerNumero = int(input("Ingresa el valor del primer numero: "))"

**Reply by IdeIsaac** [reply]
Opino lo mismo

**Comment by AlexString** [comment]
Una explicación más no está de más.

`input()` es una función de python, prácticamente la función lo que hace es pedir entrada de datos (entrada de datos en inglés es data input)

`input()` lo que recibe es una cadena de carácteres (texto) que va a ser impreso y además va a esperar a que el usuario escriba algo y presione enter para confirmar.

Entonces:
```python
input("Ingresa un numero: ")
# imprime
#Ingresa un numero: 
```
Pero para guardar el valor que recibe `input()` hay que asignarle que se guarde en una variable/objeto

```python
nombre = input("Dime tu nombre: ")
```

Si quieres guardar un valor en una variable basta con que se lo indiques:

```python
numero = 10
nombre = "Alex"

# Sin embargo guardar numeros de esta forma:
numero = "10" 
# Python lo interpretara como un valor 'cadena de carácteres' y no un valor 'numerico'

# Se soluciona con un 'casteo' de la variable, claro:
total = int(numero) *2
# total = 20
# Pero toma en cuenta que tendrás que castear la variable siempre que quieras utilizar el valor como valor numerico.

# Si quieres guardar un valor como numerico el casteo es justo como lo hiciste en la suma pero en el input():
valorNumerico = int(input("Dime tu edad: "))
```

**Reply by larap03** [reply]
gracias por tu apoyo ya lo resolví.
